lib/matplotlib/backend_managers.py

```diff
@@ -74,6 +72,44 @@ def __init__(self, canvas):
         self.keypresslock = widgets.LockDraw()
         self.messagelock = widgets.LockDraw()
 
+        self._figure = None
+        if figure:
+            self.set_figure(figure)
+
+    @property
+    def canvas(self):
+        """Canvas managed by FigureManager"""
+        return self._figure.canvas
+
+    @property
+    def figure(self):
+        """Figure that holds the canvas"""
+        return self._figure
+
+    @figure.setter
+    def figure(self, figure):
+        self.set_figure(figure)
+
+    def set_figure(self, figure, update_tools=True):
+        """
+        Sets the figure to interact with the tools
+
+        Parameters
+        ==========
+        figure: `Figure`
+        update_tools: bool
+            Force tools to update figure
+        """
+
+        if self._key_press_handler_id:
+            self.canvas.mpl_disconnect(self._key_press_handler_id)
+        self._figure = figure
+        self._key_press_handler_id = self.canvas.mpl_connect(
+            'key_press_event', self._key_press)
```

Comment:
But I think we should definitly allow the user to reset to None by explicitly passing in None, otherwise, we cannot return to the default original state.  Thus we should match the if statement from the init here.  Maybe a function ``is_bound`` or something to make this test more changeable in the future.